src/include/btmem.h

```diff
@@ -960,15 +960,77 @@ struct __wt_ref {
+     * WT_REF.del field is left in place so the read and instantiation can be repeated as necessary.
+     *
+     * Doing anything other than testing if WT_REF.ft_info.del/update is non-NULL (which eviction
+     * does), requires the WT_REF be locked.
+     *
+     * Because this is a union it is important to always access the correct field. It is also vital
+     * to interpret the state correctly and consider all the possible cases.
+     *
+     * These are the possible cases:
+     *
+     * 1. The WT_REF state is WT_REF_DELETED and ft_info.del is NULL. The page is on disk only, is
+     * deleted, is no longer accessible, and the on-disk page (if any) can be discarded freely.
+     *
+     * 2. The WT_REF state is WT_REF_DELETED and ft_info.del is not NULL. The page is on disk only,
+     * and is deleted, but the deletion may not yet be globally visible. ft_info.del describes the
+     * delete operation. If it is necessary to read the page on behalf of someone who cannot see the
+     * deletion, the page must be instantiated as described above.
+     *
+     * 3. The WT_REF state is WT_REF_DISK and ft_info.del is NULL. This is an ordinary on-disk
+     * page.
+     *
+     * 4. The WT_REF state is WT_REF_DISK and ft_info.del is not NULL. This is only permissible in
+     * readonly trees. The page is on disk only, and is deleted, but the deletion may not yet be
+     * globally visible. This differs from the case where the state is WT_REF_DISK only because the
+     * state is not set back to WT_REF_DELETED when an instantiated truncated page in a readonly
+     * tree is discarded by eviction.
+     *
+     * 5. The WT_REF state is WT_REF_MEM and the tree is read-write, and ft_info.update (not
+     * ft_info.del) is NULL. This is an ordinary in-memory page. It might have been previously
+     * deleted and then instantiated, or not, but the difference no longer matters.
+     *
+     * 6. The WT_REF state is WT_REF_MEM and the tree is read-write, and ft_info.update (not
+     * ft_info.del) is not NULL. This is a deleted page that was instantiated when the delete
+     * transaction was not yet committed. ft_info.update is the list of updates created by the
+     * instantiation, which is used to commit or abort them as needed and then cleared. It is not
+     * possible to get to this state if the truncate information was read from disk; uncommitted
+     * (including prepared) truncates are not evicted or checkpointed.
+     *
+     * 7. The WT_REF state is WT_REF_MEM and the tree is readonly, and ft_info.del (not
+     * ft_info.update) is not NULL. This is a deleted page that has been instantiated. The delete
+     * information is kept so that the page can be instantiated correctly again if it is discarded
+     * by eviction and read back in. Because the tree is readonly, it is safe to use the delete
+     * information when reading, e.g. to skip the page; no newer updates can be applied to the page.
```

Comment:
Could we simplify this by splitting into permissible states for live/read-only trees, it simplifies things? To say "ft_info.del is not NULL, but ft_info.update is NULL" is hard to understand.

Reply:
I agree it's not so great this way, but making two lists of states duplicates most of them and i don't think that's better. I will try again with the union member stuff. (In[2e5af24](https://github.com/wiredtiger/wiredtiger/pull/7909/commits/2e5af24e34f8557ee0cb39ab8a1f28f054c35c84), please have another look)